providers/Flipt/src/Cache.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace OpenFeature\Providers\Flipt;
+
+use Psr\SimpleCache\CacheInterface;
+
+class Cache {
+
+    protected const CACHE_KEY = 'openfeature-flipt';
+
+    protected CacheInterface|null $cache;
+
+
+
+    public function __construct( CacheInterface $cache = null ) {
+        $this->cache = $cache;
+    }
+
+
+
+    /**
+     * Retrievies a value from the cache
+     */
+    public function get( $key ) {
+        
+        if( empty( $this->cache ) ) return null;
+
+
+        $entries = $this->cache->get( self::CACHE_KEY, [] );
+
+        if( array_key_exists( $key, $entries ) ) return $entries[ $key ];
+
+        return null;
+    }
+
+
+    /**
+     * Sets the $value into the cache
+     */
+    public function set( $key, $value ) {
+
+        if( empty( $this->cache ) ) return;
+
+        $entries = $this->cache->get( self::CACHE_KEY, []);
+        $entries[ $key ] = $value;
+
+        $this->cache->set( self::CACHE_KEY, $entries );
```

Comment:
This introduces race conditions for distributed storage caches like Redis or Memcached. You can atomically interact with the cache per key if you utilize a key prefix instead.


```suggestion
        $this->cache->set($key, $value);
```

This does not fix the issue I mentioned below about the collisions. To resolve both, you would still need a different solution.

Reply:
Or, utilize a sufficiently large range hash function to avoid collisions. This might be something to consider being a configurable setting, e.g. raw JSON encoded keys or md5 or sha256 would all be valid strategies, to give you the overall safety necessary for the application.

Reply:
Now this is the hard part. With PSR-16 there is no way to delete all keys with a certain cache key prefix.
So I have to ways to solve that either the way you have seen ( all entries under one key ) or I store the keys to delete under a separate cache key. But also there you have racing conditions as you mentioned. But the later is probably more efficient.

I actually dont know how to solve that with PSR-16. Maybe just by documenting that this is not ready to use with a distributed caching storage.

Reply:
Dedicated cahce storage -> clear will clear all keys.